tests/test_applets/batchProcessing/test_batchProcessingGui.py

```diff
@@ -0,0 +1,203 @@
+from unittest import mock
+
+import pytest
+from ilastik.applets.batchProcessing.batchProcessingGui import (
+    BatchProcessingDataConstraintException,
+    BatchProcessingGui,
+    BatchRoleWidget,
+    FileListWidget,
+)
+from PyQt5.QtCore import Qt, QUrl
+
+
+def construct_widget(qtbot, widget_class, *args, **kwargs):
```

Comment:
This could be made marginally simpler if callers create widgets themselves: `construct_widget(qtbot, BatchRoleWidget("TestRoleName"))`. Also, maybe `prepare_widget` would be more descriptive?